awesome_owl/static/src/todo/todoList.js

```diff
@@ -0,0 +1,33 @@
+import { Component, useState } from "@odoo/owl"
+import { useAutofocus } from "@web/core/utils/hooks";
+import { Todo } from "./todo"
+
+export class TodoList extends Component {
+    static template = "awesome_owl.TodoList"
+    static components = { Todo }
+    numAllocatedIds = 0;
+
+    setup() {
+        this.todoAddInputRef = useAutofocus({ refName: 'todoAddInputRef' });
+        this.state = useState({
+            todos: [],
+        });
+    };
+
+    addTodo(event) {
+        if (event.keyCode === 13 && event.target.value !== "") {
+            this.numAllocatedIds++;
+            const newTodo = { description: event.target.value, id: this.numAllocatedIds, isCompleted: false };
+            this.state.todos.push(newTodo);
+            event.target.value = "";
+        };
+    };
+
+
+    toggleCompletionState(todoId) {
+        const todoIndex = this.state.todos.findIndex((todo) => todo.id == todoId);
+        const initialState = this.state.todos[todoIndex].isCompleted;
+
+        this.state.todos[todoIndex].isCompleted = !initialState;
```

Comment:
A possible better way?:
```suggestion
        const todo = this.state.todos.find(({ id }) => id === todoId);
        if (todo) {
            todo.isCompleted = !todo.isCompleted;
        }
```